### DISCUSSION TITLE
Connection Pool behavior on read and write queries

### DISCUSSION THREAD

**Discussion by pipaman** [vitaly-t/pg-promise#883] [General]
Hi,
I am having some issues when there are several clients through a pool of 50-100 clients. When I have a writer, it seems that I need to wait until all the readers finish.
Is there a concept of queue that I can use in some way so the writer only waits until the pending queries finish but it is executed before new coming read queries?

Thanks.

**Comment by vitaly-t** [comment]
You are probably using a transaction for writing. In this case, you should research non-blocking transactions for PostgreSql. Or, maybe you are using transactions for read operations by mistake. Cannot speculate further without seeing the code.

The bottom line, this library only executes queries and transactions as they are, it does not add anything synthetic on top. So your question is then more about PostgreSql rather than this library.

**Comment by pipaman** [comment]
Do you mean that if I use a pool on pg-promise, the task is submitted and depends exclusively on Postgres? I mean, there is no a try to execute task that fails and your code try to do it again and again
